SetupBench migration + Facebook-zstd experiment

SamLovesHoneyWater opened the pull request:
This pull request introduces a comprehensive benchmarking framework for evaluating environment setup using Dockerfiles and JSON-based rubrics. It adds scripts, documentation, example Dockerfiles, rubrics, and sample reports to enable automated validation of environment setup for various repositories. The most important changes are grouped below:

**Benchmarking Framework & Documentation**

* Added `benchmark/README.md` documenting the setup, usage, rubric format, supported test types, error handling, and output structure for the Dockerfile evaluation framework.
* Added `benchmark/requirements.txt` clarifying that the evaluation script uses only Python's standard library and requires Docker.

**Artifacts & Rubrics**

* Added example Dockerfiles (`artifacts/example.dockerfile`, `artifacts/facebook_zstd.dockerfile`, `artifacts/facebook_zstd_generic.dockerfile`) that fulfill the requirements specified in their respective rubric JSON files. These Dockerfiles install necessary tools, set environment variables, and prepare project files for evaluation.
* Added sample rubric JSON files (`rubrics/example.json`) specifying test cases and scoring for environment validation.

**Automation Scripts**

* Added `down_all.sh` script for automated cloning of a large set of benchmark datasets from GitHub into the `data` directory.

**Sample Reports**

* Added sample evaluation reports (`reports/example_report.json`, `reports/facebook_zstd_report.json`) showing the output format, test results, scores, and execution times for the benchmarked Dockerfiles.

SamLovesHoneyWater committed:
setupbench repo migrate

SamLovesHoneyWater committed:
facebook

Commit:
Successful facebook_zstd experiment

EaminC commented:
1. There are some empty file name in dir check command
<img width="285" height="167" alt="image" src="https://github.com/user-attachments/assets/20e27009-15a8-4533-a764-176d55feec0d" />
How to reproduce:
download data/facebook_zstd
python DockerfileEvaluator.py --docke
rfile artifacts/facebook_zstd.dockerfile --repo facebook_zstd --output reports/example_report.json --verbose
Starting evaluation for repo: facebook_zstd


2.Inalignment of test numbers
in case 1, the total test is 38
<img width="785" height="219" alt="image" src="https://github.com/user-attachments/assets/3c1c5530-b084-4a0d-8d2e-94d786b391f9" />
in other case,the number is 61
<img width="387" height="178" alt="image" src="https://github.com/user-attachments/assets/9d95113b-e464-4a4d-8aef-f9b61eac1750" />
I think there is actually more test
Can you have a check or tell me the reason of discarding the rest?
https://github.com/EaminC/EnvGym/blob/main/EnvBench/scripts/facebook_zstd/envbench.sh

3. The evalutator runs fail on other dockerfiles generate by other method
https://github.com/EaminC/ENVGYM-baseline
3.1 Add a path parameter for --dockerfile ,i.e. parse the parameter of the path of dockerfile
3.2 Make Sure it can run on other 11 dockerfile(currently test codex/gpt41 claude/opus4,exit with 0 result and exit code 1)

SamLovesHoneyWater committed:
Fixed list-style inputs; Added better logging.

SamLovesHoneyWater committed:
batch eval

SamLovesHoneyWater commented:
1. Fixed
2. This behavior is expected. Some tests are given more weight "scores" to reflect more importance.

SamLovesHoneyWater committed:
Fix bugs

SamLovesHoneyWater committed:
ignore reports

SamLovesHoneyWater committed:
gitigonore

SamLovesHoneyWater commented:
Added 3. batch eval script to test one repo on all versions of dockerfiles.

EaminC commented:
2.
<img width="313" height="45" alt="image" src="https://github.com/user-attachments/assets/b9e7d33a-6a72-41cb-a18b-bbd79681768c" />

Can you tell me the reason of shorten the test numbers from 71 to 38, i expected there should be more

fix on 1,3: will test

SamLovesHoneyWater committed:
FIxed path copy issue, Enhanced facekbook zstd

SamLovesHoneyWater committed:
missing import "re"

SamLovesHoneyWater committed:
rubric refinement

SamLovesHoneyWater commented:
> <img alt="image" width="313" height="45" src="https://private-user-images.githubusercontent.com/84763321/495685017-b9e7d33a-6a72-41cb-a18b-bbd79681768c.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTkzMDM2NTAsIm5iZiI6MTc1OTMwMzM1MCwicGF0aCI6Ii84NDc2MzMyMS80OTU2ODUwMTctYjllN2QzM2EtNmE3Mi00MWNiLWExOGItYmJkNzk2ODE3NjhjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEwMDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMDAxVDA3MjIzMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWVjMmI1NDM5ZDZmNDQ5YTAzNGRlMmZmZDZhMzZjYzk4OGRkNGJlNWZhY2Y2ZDhjMzQxN2I2MTY0ZWIwYjNjYjMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.F6QKUtjcfkLYwiGJSNSylAL0T8H6fQMifyz-QEH6j8U">
> Can you tell me the reason of shorten the test numbers from 71 to 38, i expected there should be more
> 
> fix on 1,3: will test

In our new rubric schema, to reduce redundant code, simple tests can be combined into one single test. For example, to check build tools:
```
{
      "id": "check_build_tools",
      "type": "commands_exist",
      "params": {
        "names": ["gcc", "g++", "make", "cmake", "ninja", "pkg-config"]
      }
}
```
This would be 6 different tests in our original pipeline. Please check all "xxx_exist" statements in the new rubric, it has reduced our total number of tests by 29. Optionally, each of those large composite tests can be given a larger weight by setting the "score" property.

There are a handful of tests that were actually missing / unaligned, I have added / modified them.

SamLovesHoneyWater committed:
Correct number of tests

SamLovesHoneyWater committed:
Copy source repo to dockerfile parent directory

SamLovesHoneyWater committed:
Remove temporary dockerfile

SamLovesHoneyWater committed:
Supports both "inside" and "outside" dockerfiles

SamLovesHoneyWater commented:
Updates:
- Number of tests reported now report combined tests as separate tests
- Copies source repo to dockerfile AND copies dockerfile itself to satisfy two scenarios:
    1. Dockerfile expects to be in the same parent directory as the repo source code
    ```
    folder/
    ├── envgym.dockerfile
    └── repo_name/
        ├── repo file 1
        ├── repo file 2
        └── xxxx
    ```
    2. Dockerfile expects to be inside the repo source code directory
    ```
    folder/
    └── repo_name/
        ├── envgym.dockerfile
        ├── repo file 1
        ├── repo file 2
        └── xxxx
    ```
To cater for BOTH scenarios, the test-time repo is copied with the following structure:
```
folder/
└── repo_name/
    ├── envgym.dockerfile
    ├── repo file 1
    ├── repo file 2  
    ├── xxxx
    └── repo_name/
        ├── repo file 1
        ├── repo file 2
        └── xxxx
```

SamLovesHoneyWater committed:
debug

SamLovesHoneyWater committed:
report by repo bug

SamLovesHoneyWater committed:
Fix our methods failing issue

SamLovesHoneyWater committed:
safe deletion of existing files

SamLovesHoneyWater committed:
print docker build command

SamLovesHoneyWater committed:
Extend timeout

SamLovesHoneyWater committed:
build err logging

SamLovesHoneyWater committed:
rollback

SamLovesHoneyWater committed:
log docker output

SamLovesHoneyWater committed:
longer timeout

SamLovesHoneyWater committed:
longer timeout

SamLovesHoneyWater committed:
down_test

SamLovesHoneyWater committed:
remove type annot

SamLovesHoneyWater commented:
Fixed problems on facebook_zstd rubric over all 11 methods:
1. Added proper path copy and clean up
2. Branched repo/envgym/dockerfile vs repo/dockerfile scenarios
3. Verified performance in line with baseline
